src/Unosquare.Swan/Components/ObjectMap.cs

```diff
@@ -0,0 +1,125 @@
+﻿namespace Unosquare.Swan.Components
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Linq.Expressions;
+    using System.Reflection;
+
+    /// <summary>
+    /// Represents an object map
+    /// </summary>
+    /// <typeparam name="TSource">The type of the source.</typeparam>
+    /// <typeparam name="TDestination">The type of the destination.</typeparam>
+    /// <seealso cref="Unosquare.Swan.Components.IObjectMap" />
+    public class ObjectMap<TSource, TDestination> : IObjectMap
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ObjectMap{TSource, TDestination}" /> class.
+        /// </summary>
+        /// <param name="intersect">The intersect.</param>
+        public ObjectMap(IEnumerable<PropertyInfo> intersect)
+        {
+            SourceType = typeof(TSource);
+            DestinationType = typeof(TDestination);
+
+            Map = new Dictionary<PropertyInfo, List<PropertyInfo>>();
+
+            foreach (var property in intersect)
+            {
+                Map.Add(DestinationType.GetProperty(property.Name), new List<PropertyInfo> { SourceType.GetProperty(property.Name) });
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the map.
+        /// </summary>
+        public Dictionary<PropertyInfo, List<PropertyInfo>> Map { get; }
+
+        /// <summary>
+        /// Gets or sets the type of the source.
+        /// </summary>
+        public Type SourceType { get; }
+
+        /// <summary>
+        /// Gets or sets the type of the destination.
+        /// </summary>
+        public Type DestinationType { get; }
+
+        /// <summary>
+        /// Maps the property.
+        /// </summary>
+        /// <typeparam name="TDestinationProperty">The type of the destination property.</typeparam>
+        /// <typeparam name="TSourceProperty">The type of the source property.</typeparam>
+        /// <param name="destinationProperty">The destination property.</param>
+        /// <param name="sourceProperty">The source property.</param>
+        /// <returns>
+        /// An object map representation of type of the destination property 
+        /// and type of the source property
+        /// </returns>
+        public ObjectMap<TSource, TDestination> MapProperty
+            <TDestinationProperty, TSourceProperty>(
+                Expression<Func<TDestination, TDestinationProperty>> destinationProperty,
+                Expression<Func<TSource, TSourceProperty>> sourceProperty)
+        {
+            var memberDestinationExpression = destinationProperty?.Body as MemberExpression;
```

Comment:
I would add null checks here instead of using ?. shortcut here. Nulls are exceptional cases should be explicit as possible

Reply:
👍